### DISCUSSION TITLE
Moose-based application capabilities

### DISCUSSION THREAD

**Discussion by robfairh** [idaholab/moose#21290] [Q&A Modules: General] [answered]
Hi all,

I have a few questions about moose in general and its applications.
I am looking for a module/tool/suite of tools with the following capabilities:
1. Irradiation/Depletion/Burnup
2. Delayed gamma heating (gamma transport + gamma heat deposition)
3. Activation heating (heating from proton/beta/alpha decays)
4. Conjugate heat transfer with natural circulation (not-Boussinesq)
5. Turbulence modeling (for the moment, RANS should suffice)
6. Radiation heat transfer
7. Machine learning algorithms for reduced order modeling, such as feed-forward neural networks and LSTMs.

Thank you for your time.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I wont comment extensively on 1-2-3, but you should request access to Griffin for some of these capabilities. 
https://inl.gov/ncrc/
4-5 if you can wait a couple months and plan to model something reasonable the MOOSE navier stokes module will be able to do it. If you plan to model large systems or go beyond RANS you should check out Nek and its moose wrapping Cardinal
https://github.com/neams-th-coe/cardinal
6 we have this capability in the heat_transfer module, using ray tracing in MOOSE
7 we can now build moose with libtorch so we can do more and more of this every day
@grmnptr can we do FFNN and LSTMs already?

Guillaume

**Comment by grmnptr** [comment]
Hey! 
Yes, we can definitely do simple ANNs, especially for ROMs.Here is a little example:
https://mooseframework.inl.gov/source/utils/LibtorchArtificialNeuralNet.html

We utilize the C++ frontend of pytorch (libtorch). At the moment, we don't have recurrent neural nets coded up but it should be easy to do. We know of people who already use these tools for deep reinforcement learning within MOOSE.

**Reply by grmnptr** [reply]
Another example on how to train a neural-net-based surrogate (ROM) from input files alone:
https://mooseframework.inl.gov/source/surrogates/LibtorchANNTrainer.html
